src/index.spec.ts

```diff
@@ -290,283 +291,273 @@ describe('EppoJSClient E2E test', () => {
       },
     );
   });
+});
 
-  describe('LocalStorageAssignmentCache', () => {
-    it('typical behavior', () => {
-      const cache = new LocalStorageAssignmentCache();
-      expect(
-        cache.hasLoggedAssignment({
-          subjectKey: 'subject-1',
-          flagKey: 'flag-1',
-          allocationKey: 'allocation-1',
-          variationKey: 'control',
-        }),
-      ).toEqual(false);
-
-      cache.setLastLoggedAssignment({
-        subjectKey: 'subject-1',
-        flagKey: 'flag-1',
-        allocationKey: 'allocation-1',
-        variationKey: 'control',
-      });
-
-      expect(
-        cache.hasLoggedAssignment({
-          subjectKey: 'subject-1',
-          flagKey: 'flag-1',
-          allocationKey: 'allocation-1',
-          variationKey: 'control',
-        }),
-      ).toEqual(true); // this key has been logged
-
-      // change variation
-      cache.setLastLoggedAssignment({
-        subjectKey: 'subject-1',
-        flagKey: 'flag-1',
-        allocationKey: 'allocation-1',
-        variationKey: 'variant',
-      });
+describe('initialization options', () => {
```

Comment:
diff is a bit hard to follow in this PR but the overview is:

* moved this block into a top level function so that it is not impacted by other tests; in particular the execution of `init` in a `beforeEach` block breaks these, since they are counting network requests
* isolated all the tests

Reply:
🙌 Heros work